File: 0882-peak-index-in-a-mountain-array/0882-peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, l: List[int]) -> int:
        s,e =0,len(l)-1
        
        while s < e:
            mid = (s+e)//2
            if l[mid] > l[mid+1]:
                e = mid
            else:
                s=mid+1

        return s

File: 0882-peak-index-in-a-mountain-array/0882-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] a) {
        int s = 1;
        int e = a.length-2;
        
        while(s<=e){
            int m = (s+e)/2;
            if(a[m]>a[m-1] && a[m+1]<a[m]){
              return m;
            }
            else if(a[m+1]>a[m]){
                s = m+1;
            }else{
                e=m-1;
            }
        }
        return 0;
    }
}